:bug: Migration related fixes

niwinz opened the pull request:
https://tree.taiga.io/project/penpot/issue/6661

superalex committed:
:bug: Fix empty components on v2 migration

superalex committed:
:bug: Fix components with compont-root on library for v2 migration

superalex committed:
:bug: Fix components with non existing component-ids for v2 migration

niwinz committed:
:sparkles: Normalize ids parsing on srepl helpers

superalex committed:
:bug: Fix conflict on components path for v2 migration

superalex committed:
:bug: Fix remove nested roots on components v2 migration

superalex commented on backend/src/app/features/components_v2.clj:
```diff
@@ -424,6 +424,20 @@
+          (letfn [(fix-container [container]
+                    (d/update-when container :objects update-vals fix-shape))
+
+                  (fix-shape [component]
```
component -> shape

superalex replied:
directly dissoc

superalex commented on backend/src/app/features/components_v2.clj:
```diff
@@ -438,6 +438,26 @@
+
+                        (cond-> shape
+                          (and (some? component-id) (nil? (ctkl/get-component (:data library) component-id)))
+                          (dissoc :component-id
```
detach-shape?

superalex replied:
if component-file exists but the library isn't attached it's valid

superalex commented on backend/src/app/features/components_v2.clj:
```diff
@@ -994,7 +995,10 @@
+        groups (->
+                (d/group-by #(first (cfh/split-path (:path %))) assets)
+                ;; If there is a group called as the generic-name we have to preserve it
+                (rename-keys {generic-name (str generic-name " " (uuid/next))}))
```
Suffix

niwinz committed:
:sparkles: Add support for `::sql/order-by` on db/sql layer

niwinz committed:
:sparkles: Improve snapshot related internal API

This commit also adds the ability to take snapshot of all files
of the team in a single run/transaction.

hirunatan commented on backend/src/app/features/components_v2.clj:
```diff
@@ -995,6 +996,9 @@
         groups (d/group-by #(first (cfh/split-path (:path %))) assets)
+        ;; If there is a group called as the generic-name we have to preserve it
+        unames (into #{} (keep str) (keys groups))
+        groups (rename-keys groups {generic-name (cfh/generate-unique-name unames generic-name)})
```
This may cause conflicts if there is an existing group with many components inside, that is split by the code below

superalex replied:
It works :)
![3](https://github.com/penpot/penpot/assets/1579633/7ada7ba0-a70a-4f4a-9447-f0d818afecf2)

niwinz committed:
:bug: Fix broken text shapes without position-data on comp-v2 migration

niwinz committed:
:sparkles: Improve error handling on websocket code

niwinz committed:
:paperclip: Move repair and validate srepl helpers to srepl.main

niwinz committed:
:sparkles: Add the ability to create a file snapshot on repair
